### PR TITLE
devstack: added explicit exit command at the end

### PR DESCRIPTION
**Pull request by sjamgade**
Without this the job remains stuck.

Actually copied it from cleanvm.yaml

### CODE REVIEW

File: jenkins/ci.opensuse.org/openstack-devstack.yaml [hunk 1]
```diff
@@ -42,6 +42,6 @@
               du $file
               exit 1
           fi
-
+          exit $ret
```
Comment (aplanas) [inline jenkins/ci.opensuse.org/openstack-devstack.yaml]: Why is this needed?

**Comment by Itxaka** [comment]
Commit message should be more explicit on how and why this fixes it.

File: jenkins/ci.opensuse.org/openstack-devstack.yaml (hunk 1 above)
Comment (jdsn) [inline jenkins/ci.opensuse.org/openstack-devstack.yaml]: Please provide a minimal reproducer to prove that this is needed.

Comment (bmwiedemann) [inline jenkins/ci.opensuse.org/openstack-devstack.yaml]: I think, it could get stuck in line 31 if one of the services started by qa_devstack.sh does not properly daemonize (detach from the console / process group)

**Comment by sjamgade** [comment]
There is a full devstack deployement that keep running after the script is done. But all those processes run under a separate user `stack`, I am not sure jenkins considers those processes as build steps and waits for them to finish, causing the job to keep on running.

Apart the actual return value is `$ret` and not from the execution of `if` condition

**Comment by bmwiedemann** [comment]
But $ret is always 0 at that point, because everything else would have triggered the `if` above.

**Comment by bmwiedemann** [comment]
Found a workaround: killing all swift processes helped getting ssh to finish.

```
killall swift-object-auditor swift-container-auditor swift-account-replicator swift-account-reaper swift-container-sync swift-object-updater swift-container-replicator swift-account-server swift-container-server swift-object-server swift-proxy-server swift-object-reconstructor swift-object-replicator swift-account-auditor swift-container-updater
```

thus we need to check+fix the script that starts those.
